src/connections.rs

```diff
@@ -43,76 +39,71 @@ impl Connection {
         self.quic_conn.remote_address()
     }
 
-    /// Send message to peer and await for a reponse.
-    pub async fn send(&mut self, msg: Bytes) -> Result<Bytes> {
+    /// Get connection streams for reading/writing
+    pub async fn create_connection_streams(&self) -> Result<ConnectionStreams> {
         let (send_stream, recv_stream) = self.quic_conn.open_bi().await?;
-        self.send_msg(send_stream, msg).await?;
-
-        trace!(
-            "Awaiting for response from remote peer: {}",
-            self.quic_conn.remote_address()
-        );
-
-        // Let's read the response (we expect one single message)
-        let received_bytes = recv_stream.read_to_end(self.max_msg_size).await?;
-        trace!(
-            "Received {} bytes from remote peer: {}",
-            received_bytes.len(),
-            self.quic_conn.remote_address()
-        );
-
-        match WireMsg::from_raw(received_bytes)? {
-            WireMsg::UserMsg(msg_bytes) => Ok(Bytes::copy_from_slice(&msg_bytes)),
-            msg => {
-                warn!("Unexpected message type received: {:?}", msg);
-                Err(Error::UnexpectedMessageType)
-            }
-        }
+
+        Ok(ConnectionStreams::new(
+            self.remote_address(),
+            send_stream,
+            recv_stream,
+            self.max_msg_size,
+        ))
+    }
+
+    /// Send message to peer and await for a reponse.
+    pub async fn send(&self, msg: Bytes) -> Result<ConnectionStreams> {
+        let mut connection_streams = self.create_connection_streams().await?;
+        connection_streams.send(msg).await?;
+        Ok(connection_streams)
     }
 
     /// Send message to peer using a bi-directional stream without awaiting for a reponse.
     pub async fn send_only(&self, msg: Bytes) -> Result<()> {
```

Comment:
I'm wondering if we should just remove this function now, if the user wants to only send on a bi-stream then `send` can be used and the result returned can be simply dropped by the user obtaining the same result?

Reply:
That'd still give you unnecessary wait time, no?

For `send` to work, you'd have to `await` the response, regardless of if you're going to use it?

Reply:
That's changed now. send will return the streams so you can do `recv.read()` if you want the response or just drop them if you don't expect a response.